fix: updating all-cabal-hashes do not invalidate callHackage (#194753)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2211.section.md

- The `nodePackages` package set now defaults to the LTS release in the `nodejs` package again, instead of being pinned to `nodejs-14_x`. Several updates to node2nix have been made for compatibility with newer Node.js and npm versions and a new `postRebuild` hook has been added for packages to perform extra build steps before the npm install step prunes dev dependencies.

> - `haskellPackage.callHackage` is not always invalidated if `all-cabal-hashes` changes, leading to less rebuild of haskell dependencies.

Reviewer: ```suggestion
- `haskellPackage.callHackage` is not always invalidated if `all-cabal-hashes` changes, leading to less rebuilds of haskell dependencies.
```

> - `haskellPackages.callHackage` and `haskellPackages.callCabal2nix` (and related functions) do not keep anymore a reference to the `cabal2nix` call used to generate them. They will be garbage collected more often.

Reviewer: ```suggestion
- `haskellPackages.callHackage` and `haskellPackages.callCabal2nix` (and related functions) no longer keep a reference to the `cabal2nix` call used to generate them. As a result, they will be garbage collected more often.
```
